types/collection.d.ts

```diff
@@ -41,6 +44,7 @@ export declare class CollectionAPI {
     listCollections(options: APIOptions): Promise<Collection[]>;
     createCollection(payload: CollectionCreate): Promise<Collection>;
     getCollection(id: string): Promise<Collection>;
+    getEagerCollection(id: string): Promise<Omit<Collection, "products" | "streamers"> & { streamers: Streamer[], products: ProductCollection[] }>;

```

Comment:
Didn't mean to mislead you, but what I meant was doing what you just did (using `Omit`) but keeping the interface you had (`EagerCollection`?)

Reply:
Oh, got it!